src/List/Zipper.elm

```diff
@@ -153,6 +153,16 @@ mapAfter f (Zipper ls x rs) =
     Zipper ls x (f rs)
 
 
+{-| Apply a function to the focussed element and another function to elements without focus.
+-}
+mapBoth : { mapFocus : a -> b, mapNoFocus : a -> b } -> Zipper a -> Zipper b
```

Comment:
I would have expected `mapBoth` to map both before and after.
Maybe calling this `transform` or changing it to be:
`transform : (Position a -> b) -> Zipper a -> Zipper b`
where `Position` is:
```elm
type Position a
  = Before a
  | Current a
  | After a
```
In case we think keeping separate functions for the transformations I would prefer to have `{ mapCurrent : a -> b, mapBefore: a -> b, mapAfter: a -> b }`.

Reply:
Hey, thanks for the input. Have you had a look at issue #27? This is meant to be a *shorthand* to `map` over all values, but with a different result based on the element being selected/focussed or not.

I'm not sure about naming and open to proposals. In my opinion `transform` wouldn't be consistent with `map`.

Reply:
I think I agree with @stoeffel that `{ mapCurrent : a -> b, mapBefore: a -> b, mapAfter: a -> b }` offers a bit more flexibility. The additional overhead for @andys8 usecase (who only needs a `mapNotCurrent`) is minimal and acceptable I think.

I could also live with a a separate type `Position`. Each of the proposed functions could all be rewritten in terms of one of the others I believe.

As @andys8 has concrete need for such a function and prefers `mapBoth`, let's go with that (but preferably with a `mapBefore` and `mapAfter`).

Reply:
Feel free to decide and tell me your preferred function signature and name. Then I'll change it :)

Reply:
Let's go with

```elm
mapAll : { mapCurrent : a -> b, mapBefore: a -> b, mapAfter: a -> b } -> Zipper a -> Zipper b
```

(as I think that the "both" in `mapBoth` doesn't apply anymore 😉 )